libpod/container_internal_common.go

```diff
@@ -1255,6 +1255,29 @@ func (c *Container) restore(ctx context.Context, options ContainerCheckpointOpti
 	c.state.RestoreLog = path.Join(c.bundlePath(), "restore.log")
 	c.state.CheckpointPath = c.CheckpointPath()
 
+	if options.IgnoreStaticIP || options.IgnoreStaticMAC {
+		networks, err := c.networks()
+		if err != nil {
+			return nil, 0, err
+		}
+
+		for net, opts := range networks {
+			if options.IgnoreStaticIP {
+				opts.StaticIPs = nil
+			}
+			if options.IgnoreStaticMAC {
+				opts.StaticMAC = nil
+			}
+			// TODO: we should properly make this in one db call instead to prevent partial states on interupt
+			if err := c.runtime.state.NetworkDisconnect(c, net); err != nil {
+				return nil, 0, fmt.Errorf("failed to rewrite network config: %w", err)
+			}
+			if err := c.runtime.state.NetworkConnect(c, net, opts); err != nil {
+				return nil, 0, fmt.Errorf("failed to rewrite network config: %w", err)
+			}
```

Comment:
@mheon PTAL, should I create a function to make this in a single db call?

Reply:
Is the intent here just to mutate the options of the given network? If so, a `NetworkModify` call could be added. I can't think of any adverse side effects for doing so...

That said, is there a reason this has to be done here, instead of setting the options correctly when we first restore the container / add it to the DB?

Reply:
> That said, is there a reason this has to be done here, instead of setting the options correctly when we first restore the container / add it to the DB?

podman container checkpoint/restore keeps the cotnainer in the db, so we have to change the db content
If you look at the code I removed, the `restore --import` path already ignores the mac/ip correctly, moving it here makes sure all cases work as expected.